### ISSUE TITLE
Hy terminal seems to get blocked up by async printing

### ISSUE THREAD

**Issue by riatzukiza** [hylang/hy-mode#73]
When using the hy command as a terminal with async operations, soon as something is printed, I cannot enter any input to hy with out with out escaping out, like there is an incomplete expression

```lisp
;; ...sent in a bunch of text with emacs funcions, some are async events.
=> ... ... ... ... => => ... ... ... => => Logged in as
Duck
449279570445729793
(print "yo") ;; I try running another command after the async



  C-c C-c=> (print "yo")
yo
=> 
```

**Comment by riatzukiza** [comment]
I meant to write this to https://github.com/hylang/hy/issues

**Comment by tsukimizake** [comment]
Hi. I tried hy-mode and botherd by freeze when opening .hy file. In most cases, the freeze can be dealt with C-c, but sometimes whole emacs gets frozen.
I figured out that loading many lines of code to hy repl at once results in broken input like 

> (set(set(set(setsetv monty 1)

, and it is parsed as if there should be more close parens.

Seems like it's a problem of hy repl (or python repl? according to the discussion in https://github.com/hylang/hy/issues/1628), but maybe we can avoid it by splitting input to chunks or something.

My environment is OSX 10.14.4, python 3.7.3, hy 0.16.0, hy-mode 20190310.1721. This issue also reproduces on my friend's Arch linux, only when using virtualenv.
